modules/i2c/slavex.inc.h

```diff
@@ -0,0 +1,19 @@
+/** @defgroup i2c I2C
```

Comment:
This could be merged into `i2cs.h`, for consistency with the UART module (at least).

See `uart.h` for an example to define the `i2cs_t` values (which are not singletons by the way).
The `i2cs_register_callback()` method should take an `i2cs_t` as first parameter.
If it's not an issue, you should define a global `i2cs_init()` method that will initialize all slaves (again, as in UART module).
Otherwise, make it take an `i2cs_t` as parameter.

Having a parameter instead of multiple function names allows simpler generic code: the only variable is the `i2cs_t` structure (instead of a bunch of method pointers).